base/libgit2/repository.jl

```diff
@@ -76,16 +76,25 @@ end
 """ Returns a found object """
 function revparse(repo::GitRepo, objname::AbstractString)
     obj_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
-    err = ccall((:git_revparse_single, :libgit2), Cint,
-            (Ptr{Ptr{Void}}, Ptr{Void}, Cstring), obj_ptr_ptr, repo.ptr, objname)
-    err != 0 && return nothing
+    @check ccall((:git_revparse_single, :libgit2), Cint,
+           (Ptr{Ptr{Void}}, Ptr{Void}, Cstring), obj_ptr_ptr, repo.ptr, objname)
     return GitUnknownObject(repo, obj_ptr_ptr[])
 end
 
 """ Returns id of a found object """
 function revparseid(repo::GitRepo, objname::AbstractString)
-    obj = revparse(repo, objname)
-    obj === nothing && return GitHash()
+    local obj
+    try
+        obj = revparse(repo, objname)
```

Comment:
How frequently does this get called? a try-catch might be fairly expensive. Is there a specific return code for "not found" in `git_revparse_single` ? I'd prefer returning a Nullable in the not found case if there's a specific return code for it, and throwing any other types of errors.